### PR TITLE
Monospace support, refactoring and new command

### PR DESCRIPTION
**Pull request by emawind84**
Added Monospace font support
New npczone command to link market to beacon `/npczone add name size`
Cleanup and refactoring
Changed quantity format on /pricelist result remove 4 digits after decimal point (do we need them?)

### CODE REVIEW

**Comment by jpcsupplies** [comment]
> 
> 
> why using decimal point for stock?

Ores use decimals in the game, its annoying but it was the simplest solution to avoid stock going missing, Can be safely rounded out in the lcd display however

File: Economy/Data/Scripts/Economy.scripts/EconConfig/EconDataManager.cs [hunk 1]
```diff
@@ -993,8 +997,6 @@ public static void CreatePlayerMarket(ulong accountId, long entityId, double siz
                         item.IsBlacklisted = true;
                 }
             }
-
-            EconomyScript.Instance.Data.Markets.Add(market);
```
Comment (jpcsupplies) [inline Economy/Data/Scripts/Economy.scripts/EconConfig/EconDataManager.cs:997]: hmm probably should have pointed this out before accepting the request, but the summary page wasnt obvious about this change for some reason.

Unless you changed how this works - wont removing this prevent newly detected items actually being added to the market table?

File: Economy/Data/Scripts/Economy.scripts/EconConfig/EconDataManager.cs [hunk 2]
```diff
@@ -1046,31 +1048,9 @@ private static void ValidateAndUpdateData(EconDataStruct data, List<MarketItemSt
...
-                        EconomyScript.Instance.ServerLogger.WriteVerbose("MarketItem Adding Default item: {0} {1}.", defaultItem.TypeId, defaultItem.SubtypeName);
-                    }
-                    else
-                    {
-                        // Disable any blackmarket items.
-                        if (defaultItem.IsBlacklisted)
-                            item.IsBlacklisted = true;
-                    }
-                }
+                AddMissingItemsToMarket(market, defaultPrices);
```
Comment (jpcsupplies) [inline Economy/Data/Scripts/Economy.scripts/EconConfig/EconDataManager.cs:1053]: hello line 1053

**Comment by jpcsupplies** [comment]
Not entirely sure the idea of npc stations linked to beacons is wise, but it would mean modders and server hosts could create roaming NPC traders so that's potentially interesting.. but may bog things down later..

Also not sure removing that line in econdatamanager wont bite us later  although it appears you are pushing the same thing later on ( see my code comments if github lets you, github has been rather strange since microsoft took it over.)

Everything else seems logical when i glanced over your other changes.  Just testing it offline at the moment to see if anything odd happens.

Also do you have a steam name I should be adding to the contributors list on steam workshop?

Don't forget to add new command behaviour to the help section in the main economy.cs file too.   Or comment on github how the new behaviour is meant to work and I'll add it to the steam online guide manual for the mod.

If you need my attention in a hurry email me at phoenixx@phoenixx.wox.org

I don't check the hotmail address linked to github frequently as you likely noticed lol.

Cheers
-Phoenixx
________________________________

File: Economy/Data/Scripts/Economy.scripts/EconConfig/EconDataManager.cs (hunk 1 above)
Comment (emawind84) [reply Economy/Data/Scripts/Economy.scripts/EconConfig/EconDataManager.cs:997]: I call that later just refactoring

**Comment by emawind84** [comment]
yes roaming npc was my goal, testing on my server and it works pretty well, I will add some lines to these files you told me later

**Comment by jpcsupplies** [comment]
Did a quick test on my test machine, using another mod i know adds items, no errors nothing exploded, and i could list them for sale.
I am satisfied all is well so I've pushed it to public.

As the server admin where a roaming npc trade zone might have been fun I simply registered one against myself as a player in the past.  But doing it automatically has a certain appeal and its limited to admins anyway so saves a bit of effort your way.

How does it handle the beacon being blown up, just unlinks and goes offline like a players one i assume?



Not sure where you are going with your roaming NPC traders -  but here is some thing that is sort of similar you could look over for more ideas for your roaming traders:
https://github.com/jpcsupplies/Economy_mod/issues/105
[https://avatars1.githubusercontent.com/u/13939039?s=400&v=4]<https://github.com/jpcsupplies/Economy_mod/issues/105>
[ENH] Ambient Trader Ships · Issue #105 · jpcsupplies/Economy_mod<https://github.com/jpcsupplies/Economy_mod/issues/105>
I thought we had an issue logged for this already. Must have been a discussion on another issue. Ambient Trader ships - Wishlist of Basic Logic - Spawn outside of player visual range near trade sta...
github.com


I also had an enh logged for trade zones randomly spewing out a few items (not all just a few random, and maybe some credits) of stock from their catalog if their beacon is destroyed (a sort of optional pro piracy option thing)   but i cannot find the issue number for that one so it may be under another enh idea.

regards

Nathan

________________________________

**Comment by emawind84** [comment]
> How does it handle the beacon being blown up, just unlinks and goes offline like a players one i assume?
Yes unlinks and goes offline

I have a script to make ships orbit planets so I thought was a good idea to use the NPC market on ships as well.

**Pull request merged**